lib/gollum-lib/file.rb

```diff
@@ -8,16 +8,19 @@ class File
 
     class << self
 
-      # For use with self.find: returns true if the given query corresponds to the in-repo path of the BlobEntry.
-      #
-      # query     - The String path to match.
-      # entry     - The BlobEntry to check against.
-      # global_match - (Not implemented for File, see Page.path_match)
-      # hyphened_tags  - If true, replace spaces in match_path with hyphens.
-      # case_insensitive - If true, compare query and match_path case-insensitively
-      def path_match(query, entry, global_match = false, hyphened_tags = false, case_insensitive = false)
-        path_compare(query, ::File.join('/', entry.path), hyphened_tags, case_insensitive)
-      end
+    # Get a canonical path to a file.
+    # Ensures that the result is always under page_file_dir (prevents path traversal), if set.
+    # Removes leading slashes.
+    #
+    # path           - One or more String path elements to join together. `nil` values are ignored.
+    # page_file_dir  - kwarg String, default: nil
+    def canonical_path(*path, page_file_dir: nil)
```

Comment:
I am wondering whether it makes more sense for this code to live in the `GitAccess` class.

Reply:
`canonical_path` is currently a Class method, and needs to be that because we are using it when looking up files (i.e. when there is no instance of a `File` yet). We also use the same method once in `File#initialize`, but there the result is already cached. So the question is whether `canonical_path` makes more sense as a Class method on `File`, or on `GitAccess`?